### PR TITLE
Initial before Lunch

### PR DESCRIPTION
**Pull request by cacrites**

### CODE REVIEW

File: api/controllers/budgetControllers.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//         if (budget.length === 0) throw new Error();
+//         res.json(budgets);
+//     })
+//     .catch(err => res.status(422).json(err));
+// };
+
+// const budgetGetById = (req, res) => {
+//     const { id } = req.params;
+//     Budget.findById(id)
+//     .populate('title budgetAmount') //not sure about this line
```
Comment (ryanhca) [inline api/controllers/budgetControllers.js]: You would actually want to select two fields and the data you want them to be populated with here. Right now you've only selected the fields. You haven't chosen what data needs to be populated. But you're on the right track for sure! 

Also, cleanest code award! Thinks are tight and clean!

File: api/controllers/categoryControllers.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const newCategory = new Category({ title });
+    newCategory.save(newCategory, (err, savedCategory) => {
+        if(err) { res.status(500).json(err); return};
+        res.json(savedCategory);
+    });
+};
+
+const categoryGetAll = (req, res) => {
+    Category.find({})
+    .select('title')
```
Comment (ryanhca) [inline api/controllers/categoryControllers.js]: These lines should be indented

File: api/controllers/categoryControllers.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if (categories.length === 0) throw new Error();
+        res.json(categories);
+    })
+    .catch(err => res.status(422).json(err));
+};
+
+const categoryGetById = (req, res) => {
+    const { id } = req.params;
+    Category.findById(id)
+    .populate('title')
```
Comment (ryanhca) [inline api/controllers/categoryControllers.js]: indent

File: api/controllers/expenseControllers.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      if (expenses.length === 0) throw new Error();
+      res.json(expenses);
+    })
+    .catch(err => res.status(422).json(err));
+};
+
+const expenseGetById = (req, res) => {
+  const { id } = req.params;
+  Expense.findById(id)
+    .populate('amount description', '_category', ['title'], '_budget', ['title'])
```
Comment (ryanhca) [inline api/controllers/expenseControllers.js]: This is great! The syntax in the braces isn't something that I've seen so I'm hoping it works as intended.

Comment (cacrites) [reply api/controllers/expenseControllers.js]: I saw that somewhere when I was looking up the populate command.... I can't totally remember where.. I think it was maybe the mongo docs?

File: api/controllers/budgetControllers.js (hunk 1 above)
Comment (cacrites) [reply api/controllers/budgetControllers.js]: TY! I try to make it halfway presentable lol

Comment (cacrites) [reply api/controllers/budgetControllers.js]: yeah I wasn't sure how to do that.  I think that's where that syntax below came into play lol.

**Pull request closed without merging**